### ISSUE TITLE
pigpiod shutdown problem (kernel 4.1.12+)

### ISSUE THREAD

**Issue by VecH** [joan2937/pigpio#32]
```root@raspberrypi ~ # uname -a```
```Linux raspberrypi 4.1.12+ #824 PREEMPT Wed Oct 28 16:39:49 GMT 2015 armv6l GNU/Linux```

pigpiod current version, recompiled and reinstall
```make uninstall```
```make clean```
```make install```

```shutdown -h now```
on screen this error ***and computer restarted (not shutdown)***
![Shutdown error](http://rghost.ru/746Y978xc/image.png)

**Comment by joan2937** [comment]
I don't know what is happening.

I've just upgraded a Pi 2 to 4.1.12-v7+ #824 and don't appear to get the same result.

I've tried four or five times.  Twice it shutdown straight away, the other times it shutdown after 80 seconds or so.  It has not restarted.

If pigpiod receives SIGTERM it will shutdown unless the signal handling has been overriden by the user.

Check with

sudo killall -s TERM pigpiod

I'm not convinced that it's pigpiod holding up the shutdown process.
